Automattic/jetpack
# jetpack: Use browser URL API instead of polyfills

#### Changes proposed in this Pull Request:

Editors for calendly and opentable were using `url-polyfill` to polyfill
the browser URL API. But it looks like all browsers we support already
supply it natively, so let's drop the polyfill.

Donations, Stripe connection notification, and Tiled Gallery were using
a polyfill for the Node URL API, which Node itself seems to be in the
process of deprecating. Again, drop that and use the browser URL API
instead.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






Test in whichever supported browsers seem relevant:

* Add a donations block. Check that the query it sends to the /wpcom/v2/memberships/status endpoint has the expected "source" parameter.
* Click the "Connect to Stripe" button, then cancel by clicking "Return to WordPress". See that the appropriate banner appears at the top of the editor.
* Add a Tiled Gallery, using images fetched from `*.files.wordpress.com` (I think you have to manually edit the block code if you're not testing on wpcom already?). Check that the Photon URLs wind up correct.
* Try the Calendly and/or OpenTable editors too.
